src/shapes/IText/DraggableTextDelegate.ts

```diff
@@ -140,39 +140,45 @@ export class DraggableTextDelegate {
     };
     target.setSelectionStyles(styleOverride, 0, selectionStart);
     target.setSelectionStyles(styleOverride, selectionEnd, target.text.length);
+    target.dirty = true;
     let dragImage = target.toCanvasElement({
       enableRetinaScaling,
     });
     // restore values
     target.backgroundColor = bgc;
     target.styles = styles;
+    target.dirty = true;
     //  handle retina scaling and vpt
-    if (retinaScaling > 1 || !isIdentityMatrix(vpt)) {
+    if (!isIdentityMatrix(vpt)) {
       const dragImageCanvas = createCanvasElement();
-      const size = new Point(dragImage.width, dragImage.height)
-        .scalarDivide(retinaScaling)
-        .transform(vpt, true);
+      const size = new Point(dragImage.width, dragImage.height).transform(
+        vpt,
+        true
+      );
       dragImageCanvas.width = size.x;
       dragImageCanvas.height = size.y;
       const ctx = dragImageCanvas.getContext('2d')!;
-      ctx.scale(1 / retinaScaling, 1 / retinaScaling);
       const [a, b, c, d] = vpt;
       ctx.transform(a, b, c, d, 0, 0);
       ctx.drawImage(dragImage, 0, 0);
       dragImage = dragImageCanvas;
     }
-    this.__dragImageDisposer && this.__dragImageDisposer();
-    this.__dragImageDisposer = () => {
-      dragImage.remove();
-    };
     //  position drag image offscreen
     setStyle(dragImage, {
       position: 'absolute',
-      left: -dragImage.width + 'px',
+      left: `${-dragImage.width}px`,
       border: 'none',
+      transform: `scale(${1 / retinaScaling},${1 / retinaScaling})`,
     });
-    getDocument().body.appendChild(dragImage);
-    e.dataTransfer?.setDragImage(dragImage, offset.x, offset.y);
+    // wrap the image so css transform will be respected
```

Comment:
di you try adding width/heigth to the style object divided by retina scaling, and see if the div is still necessary?

Reply:
Didn't try

Reply:
done